### PR TITLE
DSD-3096: move to a single config template

### PR DESCRIPTION
**Pull request by creynolds15**
BREAKING CHANGE: new output of tamr_config_file

### CODE REVIEW

File: examples/complete/versions.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+terraform {
+  required_version = "> 0.13"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.36, !=4.0.0, !=4.1.0, !=4.2.0, !=4.3.0, !=4.4.0, !=4.5.0, !=4.6.0, !=4.7.0, !=4.8.0"
```
Comment (laferrieren) [inline examples/complete/versions.tf:6]: What is the intended version range to support, should this be `>= 3.36, <4.0` or `>4.8, < 5.0`

File: examples/complete/versions.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.6.0"
+    }
+    local = {
+      source  = "hashicorp/local"
+      version = ">= 2.4.1"
+    }
+  }
+}
```
Comment (laferrieren) [inline examples/complete/versions.tf:21]: NIT: new line

File: examples/complete/versions.tf (hunk 1 above)
Comment (creynolds15) [reply examples/complete/versions.tf:6]: Meant to satisfy constraint here: https://tamr.slack.com/archives/C015CHS3REV/p1701968622861829?thread_ts=1701966902.085689&cid=C015CHS3REV

File: examples/ephemeral-spark/versions.tf [hunk 3]
```diff
@@ -0,0 +1,17 @@
+terraform {
+  required_version = "> 0.13"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.36, !=4.0.0, !=4.1.0, !=4.2.0, !=4.3.0, !=4.4.0, !=4.5.0, !=4.6.0, !=4.7.0, !=4.8.0"
```
Comment (laferrieren) [inline examples/ephemeral-spark/versions.tf:6]: can we cleanup these versions files, it should only need to specify the provider versions in the module itself. Also not sure what is trying to be limited here in terms of versions, maybe `>4.8.0`?

File: examples/ephemeral-spark/versions.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    tls = {
+      source  = "hashicorp/tls"
+      version = ">= 3.0.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.6.0"
+    }
+  }
+}
```
Comment (laferrieren) [inline examples/ephemeral-spark/versions.tf:17]: NIT: new lines at the end of files

**Review by tparikh — APPROVED** [review]

**Pull request merged**